### ISSUE TITLE
Automated benchmarking

### ISSUE THREAD

**Issue by raver119** [eclipse/deeplearning4j#8006]
We need to set up automated performance regression testing eventually.

Base idea: 
- Legacy operations tests can be automated, since all operations are split into groups, and they share signatures within own groups.
- For new operations we'll be adding new benchmark suits during performance iterations.

All benchmarks will be using tooling introduced recently.

The only question here is: how we want to automate that? @sshepel @saudet

**Comment by saudet** [comment]
I think it's going to be considered as "integration tests" by the CI infrastructure.

**Comment by AlexDBlack** [comment]
Right, run periodically as part of CI would make sense to me. Probably weekly should be frequent enough, though no real downside to running them more frequently too.

We also need to set up a system for storing and plotting those metrics. Easy enough to do, but that has uses other than just benchmarking results... we should probably tackle that first.

**Comment by saudet** [comment]
Jenkins provides functionality to log test results in XML, for example, and inspect it later. There's probably some UI plugin to plot stuff that we can use as well. We just need to have someone take the time and integrate all that.

**Comment by agibsonccc** [comment]
Closing — this issue is over 5 years old with no recent activity. The codebase has been significantly rewritten. If this is still relevant, please open a new issue.
